src/nodes/chunk_append/exec.c

```diff
@@ -345,49 +372,67 @@ initialize_runtime_exclusion(ChunkAppendState *state)
 		.glob = &glob,
 	};
 
-	Assert(state->num_subplans == list_length(state->filtered_ri_clauses));
-
-	lc_clauses = list_head(state->filtered_ri_clauses);
-	lc_constraints = list_head(state->filtered_constraints);
-
 	if (state->num_subplans == 0)
 	{
 		state->runtime_initialized = true;
 		return;
 	}
 
 	state->runtime_number_loops++;
+
+	if (state->initial_parent_clauses != NIL)
+	{
+		/* try to exclude all the chunks using the parents clauses.
+		 * here, all constraints are true but exclusion can still
+		 * happen because of things like ANY(empty set), and NULL
+		 * inference
+		 */
+		if (can_exclude_constraints_using_clauses(state,
+												  list_make1(makeBoolConst(true, false)),
```

Comment:
So here we're passing it a list of constraints that consists of a single true value, to effectively make it check whether the `initial_parent_clauses` evaluate to true, given the plan parameter?

Reply:
Yes, except that we are checking whether initial_parent_clauses are always false (not true) for exclusion. But yes this is simply checking the parent clauses for internal contradiction given the evaluated parameter values

Reply:
Does this mean that parent clauses should evaluate to constant false after `constify_restrictinfo_params`? I wonder if we can wrirte the check explicitly like this, `constify_restrictinfo_params` and then check that every element is a false Const.

I mean, I just spent half an hour at it, and I still can't tell what `predicate_refuted_by` does if you pass clause = False, predicate = Null, and weak = true. And this is a fallback used when there are more than one predicate, which apparently isn't covered by the new tests.